mos_tests/neutron/python_tests/test_ovs_restart.py

```diff
@@ -480,3 +482,197 @@ def test_restart_openvswitch_agent_under_bat(self, prepare_openstack):
+        res = self.run_on_vm(client, keypair, client_cmd, vm_login=vm_login,
+                             vm_password=vm_pwd)
+        return res
+
+    @pytest.fixture(autouse=True)
+    def _prepare_openstack(self, init):
+        """Prepare OpenStack for scenarios run
+
+        Steps:
+            1. Update default security group
+            2. Upload the image with pre-installed iperf
+            3. Create router01, create networks net01: net01__subnet,
+            192.168.1.0/24, net02: net02__subnet, 192.168.2.0/24 and
+            attach them to router01.
+            4. Create keypair
+            5. Launch vm1 in net01 network and vm2 in net02 network
+            on different computes
+            6. Go to vm1 console and send pings to vm2
+        """
+
+        self.setup_rules_for_default_sec_group()
+        vm_image = self.create_image('/home/aallakhverdieva/images/',
+                                     'ubuntu-iperf.qcow2')
+
+        self.instance_keypair = self.os_conn.create_key(
+            key_name='instancekey')
+        zone = self.os_conn.nova.availability_zones.find(zoneName="nova")
+        hosts = zone.hosts.keys()[:2]
+
+        # create router
+        router = self.os_conn.create_router(name="router01")
+
+        # create 2 networks and 2 instances
+        for i, hostname in enumerate(hosts, 1):
+            net, subnet = self.create_internal_network_with_subnet(suffix=i)
+            self.os_conn.router_interface_add(
+                router_id=router['router']['id'],
+                subnet_id=subnet['subnet']['id'])
+            self.os_conn.create_server(
+                name='server%02d' % i,
+                availability_zone='{}:{}'.format(zone.zoneName, hostname),
+                image_id=vm_image.id,
+                flavor=2,
+                timeout=300,
+                key_name=self.instance_keypair.name,
+                nics=[{'net-id': net['network']['id']}])
+
+        # check pings
+        self.server1 = self.os_conn.nova.servers.find(name="server01")
+        self.server2 = self.os_conn.nova.servers.find(name="server02")
+        self.server2_ip = self.os_conn.get_nova_instance_ips(
+            self.os_conn.nova.servers.find(name="server02")).values()[0]
+        self.check_ping_from_vm(self.server1, self.instance_keypair,
+                                self.server2_ip, vm_login='ubuntu',
+                                vm_password='ubuntu', timeout=4 * 60)
+
+        # make a list of all ovs agent ids
+        self.ovs_agent_ids = [agt['id'] for agt in
+                              self.os_conn.neutron.list_agents(
+                                  binary='neutron-openvswitch-agent')[
+                                  'agents']]
+
+        # make a list of ovs agents that resides only on controllers
+        controllers = [node.data['fqdn']
+                       for node in self.env.get_nodes_by_role('controller')]
+        ovs_agts = self.os_conn.neutron.list_agents(
```

Comment:
Let's move `self.ovs_agent_ids = ` below and rewrite it like:
`self.ovs_agent_ids = [agt['id'] for agt in ovs_agts]`